azure/services/groups/spec.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package groups
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha4"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/converters"
+)
+
+// GroupSpec defines the specification for a Resource Group.
+type GroupSpec struct {
+	Name           string
+	Location       string
+	ClusterName    string
+	AdditionalTags infrav1.Tags
+}
+
+// ResourceName returns the name of the group.
+func (s *GroupSpec) ResourceName() string {
+	return s.Name
+}
+
+// ResourceGroupName returns the name of the group.
+// Note that it is the same as the resource name in this case.
+func (s *GroupSpec) ResourceGroupName() string {
+	return s.Name
+}
+
+// OwnerResourceName is a no-op for groups.
+func (s *GroupSpec) OwnerResourceName() string {
+	return "" // not applicable
+}
+
+// Parameters returns the parameters for the group.
+func (s *GroupSpec) Parameters(existing interface{}) (interface{}, error) {
+	if existing != nil {
+		// rg already exists, nothing to update.
+		return nil, nil
```

Comment:
What if additional tags were added?

Reply:
I don't think it would update currently

Reply:
indeed, but we weren't doing that before either: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1667/files#diff-036de5e90528343ff92467735b06ba4c22abbb0616cc3911820360bc37aea755L59 

I tried to avoid making any sort of bug fixes / behavioral changes (besides async stuff) as part of these PRs because they are already so big. How about I file a bug issue to fix it separately?

Reply:
sgtm

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1696